assets/js/blocks/cart-checkout-shared/payment-methods/test/saved-payment-method-options.tsx

```diff
@@ -0,0 +1,137 @@
+/**
+ * External dependencies
+ */
+import { render, screen } from '@testing-library/react';
+import { registerPaymentMethod } from '@woocommerce/blocks-registry';
+import * as wpData from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import SavedPaymentMethodOptions from '../saved-payment-method-options';
+
+jest.mock( '@wordpress/data', () => ( {
+	__esModule: true,
+	...jest.requireActual( '@wordpress/data' ),
+	useSelect: jest.fn(),
+} ) );
+
+// Mock use select so we can override it when wc/store/checkout is accessed, but return the original select function if any other store is accessed.
+wpData.useSelect.mockImplementation(
```

Comment:
`mockImplementation` is currently causing a TS error. Do you think it's worth investigating if we can solve this error or add `@ts-ignore` otherwise to prevent that this PR introduces a new TS error?

Reply:
Thanks Niels, I have fixed it!

Reply:
Yeah! Thanks for fixing the TS error, @opr. 🎉

<img width="356" alt="Screenshot 2023-06-27 at 18 53 36" src="https://github.com/woocommerce/woocommerce-blocks/assets/3323310/191cd68b-e4ea-4cf6-85b8-e760d47f6ac9">